src/main/java/org/wildfly/security/http/impl/SpnegoAuthenticationMechanism.java

```diff
@@ -263,7 +263,7 @@ public void evaluateRequest(HttpServerRequest request) throws HttpAuthentication
             } else {
                 log.trace("GSSContext establishing - unable to hold GSSContext so continuation will not be possible");
                 handleCallback(AuthenticationCompleteCallback.FAILED);
-                request.authenticationFailed(log.authenticationFailed(SPNEGO_NAME));
+                request.authenticationFailed(log.authenticationFailed(SPNEGO_NAME), this::sendBareChallenge);
```

Comment:
I am not quite following why this change is required.

The reason this line does not already send a challenge is because a challenge has already been sent to the client, their response to that challenge was either invalid or has resulted in a continuation required situation - either way challenging them again is not going to change the outcome as we have no way to indicate to them why their message was rejected.

Reply:
sorry, agree - continuation is not possible in this situation so sending challenge does not make sense